.devcontainer/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/devcontainers/python:3.10-bullseye@sha256:a95bd841ad4b801a5e0c4ba183aaccefd8d1e65a3dab2116aa7b385bd70135e2
+FROM mcr.microsoft.com/devcontainers/python:3.11-bullseye@sha256:105bf6a63ab025206f019a371a735fec6553db0be520030c7a2fd0e002947232
```

Comment:
Does the devcontainer need to match the Dockerfile base image? If so, we should probably have comments in both files so future us know to update both at the same time.